xbmc/cores/DllLoader/exports/CMakeLists.txt

```diff
@@ -21,7 +21,7 @@ elseif(NOT CORE_SYSTEM_NAME STREQUAL windows)
   add_custom_target(wrapper.def ALL nm ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/wrapper.dir/wrapper.c.o | grep __wrap | awk '{ printf(\"%s \", \$\$3) }' | sed \"s/___wrap_/__wrap_/g\" | sed \"s/__wrap_/-Wl,-wrap,/g\" > wrapper.def)
 
   if(CORE_SYSTEM_NAME STREQUAL android)
-    add_custom_command(TARGET wrapper.def COMMAND echo \"-l$<TARGET_FILE:${APP_NAME_LC}>\" >> wrapper.def)
+    add_custom_command(TARGET wrapper.def COMMAND echo \"-L${kodi_BINARY_DIR} -l:libkodi.so\" >> wrapper.def)
     add_dependencies(wrapper.def ${APP_NAME_LC})
```

Comment:
You could use the TARGET_FILE_DIR and TARGET_FILE_NAME generator expression. (https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html)
They have the advantage that cmake already complains if the library target does for some reason not exist. So it makes things more robust. `kodi` should always be APP_NAME_LC, otherwise it breaks builds where someone tries to rebrand.  Also I'm wondering where the kodi_BINARY_DIR comes from.